src/Compilers/CSharp/Portable/Binder/Binder_Patterns.cs

```diff
@@ -81,8 +83,9 @@ private BoundExpression BindIsPatternExpression(IsPatternExpressionSyntax node,
 
         private BoundExpression BindSwitchExpression(SwitchExpressionSyntax node, DiagnosticBag diagnostics)
         {
-            Debug.Assert(node != null);
-            Binder switchBinder = this.GetBinder(node);
+            RoslynDebug.Assert(node is { });
+            Binder? switchBinder = this.GetBinder(node);
+            RoslynDebug.Assert(switchBinder is { });
```

Comment:
>RoslynDebug [](start = 12, length = 11)

nit: `Debug.Assert` should now work (we're compiling against netcoreapp3.1 TFM which includes the annotations we need)